### ISSUE TITLE
Reproducibility of MNIST Uniaxial Dataset Results

### ISSUE THREAD

**Issue by Rrp10157** [saeedmhz/MultiRes-WNet#2]
### Description

I am trying to reproduce the MAE and MAPE results reported in the paper for the MNIST Uniaxial dataset. However, I have not been able to obtain the same values as those reported in the paper.

### What I Have Done

* Used the hyperparameters and configuration described in the paper.
* Used the training code provided in this repository.
* Evaluated the model following the repository instructions.

### Issue

The MAE and MAPE values I obtain differ significantly from the results reported in the paper.

### Questions

1. Could you share the details of the dataset used for training and evaluation?
2. Are there any preprocessing steps, data splits, or configuration settings that are not documented in the paper or repository?
3. Could you provide the exact training and evaluation scripts used to obtain the reported MAE and MAPE values for the x- and y-displacement fields?

Any guidance would be greatly appreciated.

Thank you for your help.

**Comment by saeedmhz** [comment]
Thank you for reaching out and for your interest in reproducing our work.

Unfortunately, I no longer have the exact, specific data split used for the paper's benchmarks. However, regarding preprocessing, I can confirm that outside of horizontal and vertical mirroring (data augmentation), there weren't any other noteworthy steps that should significantly impact the training results.

To help narrow down the discrepancy, could you provide a bit more context on your current setup?

- Can you confirm whether you used the standard MNIST Uniaxial dataset and not the MNIST Fashion Uniaxial variant? The standard MNIST Uniaxial dataset is relatively simple; a basic model should achieve decent results. If your metrics are off by a large margin, there might be a subtle bug in your training pipeline or data loading.

- Which of the four model architectures available in train.py did you select for your training run?

- Could you share the exact MAE and MAPE values you are currently obtaining for both the x- and y-displacement fields so I can see how far off they are from the paper?

**Comment by Rrp10157** [comment]
Hello,

Thank you for your response.

I used the **standard MNIST Uniaxial** dataset from the OpenBU dataset. As input, I used the **MNIST_input_files**, and as the target labels, I used **FEA_displacement_results_step12**.

For training, I selected the **MultiResUNet** model architecture.

I also used the same hyperparameters as reported in the paper:

* Learning rate: **0.01 (10⁻²)**
* Learning rate decay factor: **0.5**
* Learning rate reduced every **50 epochs**
* Total epochs: **200**
* Batch size: **32**
* First-layer channels: **64** (approximately **2M parameters**)

The command I used for training is:

```bash
python train.py -lr 0.01 -d 0.5 -s 50 -e 200 -b 32 -c 64
```

The evaluation metrics I obtained are:

* **Y-displacement**

  * MAE: **0.145**
  * MAPE: **0.2168%**

* **X-displacement**

  * MAE: **0.0135**
  * MAPE: **1.5681%**


I would greatly appreciate any suggestions on what I should verify or whether there are any additional implementation details that were not included in the repository.

Thank you

**Comment by saeedmhz** [comment]
Have you tried data augmentation?

**Comment by Rrp10157** [comment]
No, I haven't tried the data augmentation.

**Comment by saeedmhz** [comment]
Try vertical and horizontal mirroring of the inputs and accordingly modify the outputs in the training set. Then retrain the model with the new augmented dataset.

**Comment by Rrp10157** [comment]
Did you train separate models for the Ux and Uy displacements?

**Comment by saeedmhz** [comment]
Yes, the weights of the trained models are recorded here: https://drive.google.com/drive/folders/1CJngW03f6tHqDoukvGUWNcaWF4QGKTn6
